### PR TITLE
Add support for nested maps to has-key

### PR DESCRIPTION
**Pull request by jathak**
See https://github.com/sass/sass-spec/pull/1558

### CODE REVIEW

File: lib/src/functions/map.dart
Comment (nex3) [inline lib/src/functions/map.dart:74]:
```suggestion
  for (var key in allKeys.take(allKeys.length - 1)) {
```

Not only is this terser, it also avoids creating an unnecessary intermediate list :smiley:.

**Review by nex3 — CHANGES_REQUESTED** [review]
Please add a CHANGELOG entry as well.

File: lib/src/functions/map.dart
Comment (jathak) [reply lib/src/functions/map.dart]: Done

File: CHANGELOG.md
Comment (Awjin) [inline CHANGELOG.md:3]: This should also link to the docs.

Comment (jathak) [reply CHANGELOG.md:3]: Will do. I'm going to wait until #1076 is merged

Comment (jathak) [reply CHANGELOG.md:9]: Done

**Review by nex3 — APPROVED** [review]

**Pull request merged**
